File: src/main/java/com/example/indentory_management_system/Service/NotificationService.java
package com.example.indentory_management_system.Service;

import java.util.List;
import com.example.indentory_management_system.dto.NotificationResponseDto;

public interface NotificationService {
    List<NotificationResponseDto> getNotificationsForUser(String email);
    void markAsRead(Long id);
    void markAllAsRead(String email);
    void createNotification(String email, String title, String message);
}

File: IMS_frontend/IMS/src/app/core/services/notification.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

export interface NotificationResponseDto {
  id: number;
  userId: number;
  userEmail: string;
  title: string;
  message: string;
  isRead: boolean;
  createdAt: string;
}

@Injectable({
  providedIn: 'root'
})
export class NotificationService {
  private readonly apiUrl = 'http://localhost:8082/api/v1/notifications';

  constructor(private http: HttpClient) {}

  getNotifications(): Observable<NotificationResponseDto[]> {
    return this.http.get<NotificationResponseDto[]>(this.apiUrl);
  }

  markAsRead(id: number): Observable<void> {
    return this.http.put<void>(`${this.apiUrl}/${id}/read`, {});
  }

  markAllAsRead(): Observable<void> {
    return this.http.put<void>(`${this.apiUrl}/read-all`, {});
  }
}
